reputation/microeconomic_reputation.tex

```diff
@@ -0,0 +1,367 @@
+Let's thus define one more symbol:
+\begin{itemize}
+\item $V_L(a) = (V_{PL}(a)- V_{RG}(a)) + (V_{RL}(a)- V_{PG}(a))$ - the total value lost because of agents breaking the terms of agreement $a$.
+\end{itemize}
+and use it to rewrite the main equation one last time:
+
+\begin{equation}
+\begin{split}
+    V_G = \sum_{a \in all\_agreements}(V_{PN}(a) + V_{RN}(a) - V_L(a)) \\
+          - \sum_{A \in all\_agents\_using\_Golem}C_A
+\end{split}
+\end{equation}
+
+This equation defines few general ways to increase the $V_G$:
+
+\begin{enumerate}
+\item Increase the number of agreements
+\item Increase the average nominal value of the agreement (e.g. by providing additional capabilities, like internet connectivity or GPU access)
+\item Decrease the average value lost because of breaches of the agreements ($V_L(a)$)
+\item Decrease the cost of the participation in the Golem market (e.g. by creating better SDKs)
+\end{enumerate}
+
+Note that these ways interact with each other: e.g. if we improve 2 by implementing some features that will be hard to use, we'll also worsen 4.
+Or: the better is the average nominal value of the agreement, the more agreements we'll expect to have.
+Or - what is important from the POV of this document - if we implement complex safeguards against cheating, they might have negative impact on all the other points.
+
+Keeping that in mind, the rest of the document is aimed at the third direction from the above list.
+
+\subsection{Honest strategy}
+
+Fo an agreement $a$ between agents $A_1$ and $A_2$ let's define the "dishonesty index" of agent $A_2$ as:
+
+\begin{equation}
+    D_{A_2}(a) = \frac{V_{A_1L}(a)}{V_{A_1N}(a)}
+\end{equation}
+
+ 
+That is: if $A_1$ received everything that was agreed, $D_{A_2}(a) = 0$. If all agreement-related costs of $A_1$ are covered, but they got nothing more, $D_{A_2}(a) = 1$.
+If $A_1$ lost on the agreement, $D_{A_2} > 1$.\footnote{Values below 0 are also possible, e.g. paying more than agreed is also "against the agreement".}
+
+The value of $V_{A_1L}(a)$ (so also the dishonesty index) depends on the decision made by agent $A_1$. 
+In an usual case the "visible decision" will by caused by the algorithm implemented in the provider/requestor agent,
+but this is just an effect of the agent's "real decision" about the implementation.\footnote{In an extreme case we can imagine 
+a human operator who directly interacts with the Golem protocol. Or just imagine someone who turns off the currently rented hardware.}
+
+The agent's core decision-making process is called a "strategy". The better agent fulfills their part of the agreements, the more "honest" is the strategy.
+So, in other words: when agent $A$ signs an agreement $a$ with an agent using a dishonest strategy, the expected value of $V_{AL}(a)$ is high.
+When the other side is fully honest, $V_{AL}(a) = 0$. 
+Note that agent's intentions don't matter here - there's no difference if an agent breaks an agreement on purpose or accidentally.
```

Comment:
I have some doubts here.

First of all, if an agent breaks an agreement accidentally (e.g. due to a hardware/software/network fault), this can hardly be considered their "strategy". 
(By the way this also shows the weakness of your "strategy" to choose such a loaded terms as "honesty/dishonesty", but I digress).

Secondly, I think there is a difference with regard to the "Pascal's mugging" you mention - a machine that randomly breaks with some probability is very different from a cunning crook there.

What is true though, is that an outside observer may often **not be able to distinguish** between the two by observing the agent behaviour.

Reply:
> First of all, if an agent breaks an agreement accidentally (e.g. due to a hardware/software/network fault), this can hardly be considered their "strategy". 

I disagree. 

Agent makes a decision about quality of the hardware/software/network etc they run the provider agent on. E.g. 
* they might decide to run an expensive a provider on a high quality hardware in a rented high quality server room
* they might decide to run a cheap provider on an old laptop in their basement

To me, this is a part of the strategy.

>(By the way this also shows the weakness of your "strategy" to choose such loaded terms as "honesty/dishonesty", but I digress).

Well, I know this is a digression, but still:
* If I know there is a certain risk my computer will fail
* Nevertheless, I rent it to agents who expect the reliability

this is sort-of-dishonest? But I don't think we have an important disagreement here, I won't be defending the term "honesty". I'm pretty bad at naming things :)

> Secondly, I think there is a difference with regard to the "Pascal's mugging" you mention - a machine that randomly breaks with some probability is very different from a cunning crook there.

Yes, that's true, but for me it is quantitative difference: machine that randomly breaks is "low quality slightly dishonest strategy", cunning crook is "(possibly) high quality very dishonest strategy".

> What is true though, is that an outside observer may often **not be able to distinguish** between the two by observing the agent behaviour.

Yes. That's exactly the reason why I want to keep them together, as a single concept.

(I could imagine corner cases when even the provider doesn't know if they are acting "fair", e.g. afair we had providers who had a cron job that purged `yagna` directories and this sometimes harmed requestors, but providers didn't know this).

Reply:
So it seems we just have to agree to disagree ;)